lib/rules/no-misleading-character-class.js

```diff
@@ -181,6 +199,88 @@ module.exports = {
         const sourceCode = context.sourceCode;
         const parser = new RegExpParser();
 
+        /**
+         * Determines what offset to start a granular report, if it's straightforward to do so.
+         * @param {Character} char Individual character being reported on.
+         * @param {string} kind What kind of character report is being made.
+         * @param {Node} node Parent string node to report within.
+         * @param {Object} previousReport Kind and start offset for the previous report, if there was one.
+         * @returns {Object | null} Report offset, or null to report on the whole node.
+         * @see https://github.com/eslint/eslint/pull/17515#issuecomment-1707697186
+         */
+        function getReportStartOffset(char, kind, node, previousReport = {}) {
+            if (node.type !== "Literal" && (node.type !== "TemplateLiteral" || node.expressions.length)) {
+                return null;
+            }
+
+            // If a problematic character exists twice in the string, start each Nth search after N - 1
+            const searchStart = previousReport && previousReport.kind === kind
+                ? (previousReport.startOffset + 1)
+                : char.start - char.raw.length + 1;
+
+            const searchableLength = searchStart + char.start + char.raw.length * 2;
+            const offsetStart = node.raw.slice(0, searchableLength).indexOf(char.raw, searchStart);
+
+            return offsetStart === -1 ? null : offsetStart;
+        }
+
+        /**
+         * Determines where a report on a character should truly start.
+         * @param {number} reportStartOffset Original estimate for where the character is.
+         * @param {Character} char Individual character being reported on.
+         * @param {string} kind What kind of character report is being made.
+         * @param {Node} node Parent string node to report within.
+         * @returns {number} Adjusted start location for a character's report.
+         */
+        function getReportStartOffsetAdjusted(reportStartOffset, char, kind, node) {
+
+            // Characters that aren't started by manual \u encodings are only adjusted by 1 or 2
+            if (!char.raw.startsWith("\\u")) {
+                return reportStartOffset - (kindsForSingleCharacterAdjustment.has(kind) ? 1 : 2);
+            }
+
+            // If the character is the second of a joined pair, report on the previous character's start
+            if (kindsForReportingOnPreviousElement.has(kind) && char.raw.startsWith("\\u")) {
+                const previousElement = char.parent.elements[char.parent.elements.indexOf(char) - 1];
```

Comment:
Characters that constitute the problematic sequence don't necessarily have the same parent. For example:

```
/* eslint no-misleading-character-class: 2 */

/[\uD83D\uDC4D-\uffff]/
```

```
TypeError: Cannot read properties of undefined (reading 'indexOf')
Occurred while linting C:\projects\eslint\foo.js:3
Rule: "no-misleading-character-class"
    at getReportStartOffsetAdjusted (C:\projects\eslint\lib\rules\no-misleading-character-class.js:244:83)
    at generateReportLocation (C:\projects\eslint\lib\rules\no-misleading-character-class.js:270:47)
    at verify (C:\projects\eslint\lib\rules\no-misleading-character-class.js:345:42)
    at Literal[regex] (C:\projects\eslint\lib\rules\no-misleading-character-class.js:359:17)
    at ruleErrorHandler (C:\projects\eslint\lib\linter\linter.js:1091:28)
    at C:\projects\eslint\lib\linter\safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (C:\projects\eslint\lib\linter\safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (C:\projects\eslint\lib\linter\node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (C:\projects\eslint\lib\linter\node-event-generator.js:326:22)
```

Reply:
Just following up - will make this change. Thanks!